tasks/insights-client.yml

```diff
@@ -0,0 +1,31 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Ensure required packages are installed
+  package:
+    name: "insights-client"
+    state: present
+  when: rhc_insights_state | d('present') == "present"
+
+- name: Call insights status
+  command: insights-client --status
+  ignore_errors: true
```

Comment:
It is far better to use `failed_when` instead of ignoring all errors.  For example - is there some error code which indicates the system is already registered?
If so, then something like this:
```yaml
failed_when: __rhc_insights_status.rc not in [0, 2]
```
assuming `2` means "system already registered".

or, is there some stdout or stderr text which is printed when the system is already registered?  If so, then something like this:
```yaml
failed_when:
  - __rhc_insights_status.rc != 0
  - not __rhc_insights_status.stdout | search("system is already registered")
```

Reply:
This case is a bit different. `insights-client` has 2 different returns depending of the `legacy_upload` configuration. That means this can be 0 or 1 and depending of that configuration it means that the command has run without problems or that `insights-client` is register/unregister. 

I want to create a fix to unify this behavior, and once is done remove the `ignore_errors` in the future.